src/tribler/core/components/knowledge/rules/knowledge_rules_processor.py

```diff
@@ -137,7 +144,16 @@ def query(_start, _end):
         self.set_last_processed_torrent_id(end)
 
         duration = time.time() - start_time
-        message = f'[Batch] Processed: {processed} titles. Added: {added} tags. Duration: {duration:.3f} seconds.'
+
+        def calculate_eta():
+            processed_in_the_current_session = end - self._start_rowid_in_current_session
+            remaining = max_row_id - end
+            duration_in_tne_current_session = time.time() - self._start_time_in_current_session
+            eta = remaining * duration_in_tne_current_session / processed_in_the_current_session
```

Comment:
Typo: "tne" -> "the"

```suggestion
            duration_in_the_current_session = time.time() - self._start_time_in_current_session
            eta = remaining * duration_in_the_current_session / processed_in_the_current_session
```